models/governance/optimism/snapshot_proposals.sql

```diff
@@ -0,0 +1,526 @@
+                  ) AS VARCHAR) AS VARCHAR),
+                  ''
+                ) AS VARCHAR),
+                ''
+              ) AS VARCHAR) AS VARCHAR),
+              ''
+            ) AS VARCHAR),
+            ''
+          ) AS VARCHAR) AS VARCHAR),
+          ''
+        ) AS VARCHAR),
+        ''
+      ) AS VARCHAR)
+    ) AS proposal_description,
+    start AS start_block,
+    FROM_UNIXTIME(start) AS start_date,
+    "end" AS end_block,
+    FROM_UNIXTIME("end") AS end_date,
+    'snapshot' AS platform
+  FROM {{ source('snapshot','proposals') }}
+  WHERE
+    "space" = 'opcollective.eth'
+    AND "type" = 'approval'
+) AS p
+LEFT JOIN (
+-- Join vote data with multiple options proposals
+  SELECT
+    proposal AS proposal_id,
+    voter,
+    vp AS votingWeightage,
+    choice,
+    CASE
+      WHEN choice = '1'
+      THEN 'for'
+      WHEN choice = '2'
+      THEN 'against'
+      WHEN choice = '3'
+      THEN 'abstain'
+    END AS status
+  FROM {{ source('snapshot','votes') }}
+  WHERE
+    "space" = 'opcollective.eth'
+) AS v
+  ON p.proposal_id = v.proposal_id
+GROUP BY
+  p.proposal_id,
+  p.proposal_description,
+  p.start_block,
+  p.start_date,
+  p.end_block,
+  p.end_date,
+  p.platform
+  
+  UNION ALL
+  
+  SELECT
+  p.proposal_id,
+  'Test Proposal' AS proposal_type, -- Set the proposal type to 'Test Proposal'
+  p.proposal_description,
+  p.start_block,
+  p.start_date,
+  p.end_block,
+  p.end_date,
+  p.platform,
+  ARRAY_AGG(v.voter ORDER BY v.voter) AS voters,
+  ARRAY_AGG(v.votingWeightage ORDER BY v.voter) AS corresponding_voting_weightage,
+  ARRAY_AGG(v.choice ORDER BY v.voter) AS corresponding_choices,
+  ARRAY_AGG(v.status ORDER BY v.voter) AS corresponding_choices_name,
+  MAX(v.votingWeightage) AS highest_weighted_vote,
+  MAX_BY(v.voter, v.votingWeightage) AS highest_weighted_voter,
+  (
+    MAX(v.votingWeightage) * 100 / SUM(v.votingWeightage)
+  ) AS highest_weighted_voter_percentage,
+  SUM(CASE WHEN v.choice = '1' THEN v.votingWeightage ELSE 0 END) AS total_for_votingWeightage,
+  SUM(CASE WHEN v.choice = '3' THEN v.votingWeightage ELSE 0 END) AS total_abstain_votingWeightage,
+  SUM(CASE WHEN v.choice = '2' THEN v.votingWeightage ELSE 0 END) AS total_against_votingWeightage,
+  SUM(CASE WHEN v.choice = '1' THEN 1 ELSE 0 END) AS unique_for_votes,
+  SUM(CASE WHEN v.choice = '3' THEN 1 ELSE 0 END) AS unique_abstain_votes,
+  SUM(CASE WHEN v.choice = '2' THEN 1 ELSE 0 END) AS unique_against_votes,
+  COUNT(v.choice) AS unique_votes_count,
+  SUM(v.votingWeightage) AS total_votes_casted,
+  CASE
+  WHEN (SUM(CASE WHEN TRY_CAST(v.choice AS varchar) = '1' THEN TRY_CAST(v.votingWeightage AS DOUBLE) ELSE 0.0 END) / SUM(v.votingWeightage)) * 100 >= 50
+  THEN 'success'
+  WHEN p.end_date > CURRENT_TIMESTAMP  
+  THEN 'active'
+  ELSE 'defeated'
+ END AS proposal_status
+FROM (
+  -- Select test proposals from Snapshot platform based on specific criteria
+  SELECT
+    id AS proposal_id,
+    CONCAT(
```

Comment:
do you mind explaining the purpose of connecting all these titles and body together? Maybe having them separated is better?

Reply:
So the decoded table of agora has it combined, hence these two columns have been combined here.